src/systems/systems/src/navigation/radio_altimeter.rs

```diff
@@ -0,0 +1,722 @@
+    travel_time: Time,
+}
+
+impl TransceiverPairResponse {
+    fn new(travel_time: Time) -> Self {
+        Self { travel_time }
+    }
+
+    fn travel_time(&self) -> Time {
+        self.travel_time
+    }
+}
+
+/// This struct represents a highly simplified pair of the transceivers as used by the ALA-52B
+/// system and models some of the rudimentary physical properties like signal travel time, maximum
+/// distance and attitude cutoffs.
+#[derive(Copy, Clone)]
+pub struct Ala52BTransceiverPair {
+    alt_above_ground_id: VariableIdentifier,
+    pitch_id: VariableIdentifier,
+    bank_id: VariableIdentifier,
+
+    transmitter: AntennaInstallation,
+    receiver: AntennaInstallation,
+
+    alt_above_ground: Length,
+    pitch: Angle,
+    bank: Angle,
+}
+
+impl Ala52BTransceiverPair {
+    const ALT_ABOVE_GROUND: &'static str = "PLANE ALT ABOVE GROUND";
+    const PITCH: &'static str = "PLANE PITCH DEGREES";
+    const BANK: &'static str = "PLANE BANK DEGREES";
+
+    pub fn new(
+        context: &mut InitContext,
+        transmitter: AntennaInstallation,
+        receiver: AntennaInstallation,
+    ) -> Self {
+        Self {
+            // STATIC CG TO GROUND can be used for calibration
+            alt_above_ground_id: context.get_identifier(Self::ALT_ABOVE_GROUND.to_owned()),
+            pitch_id: context.get_identifier(Self::PITCH.to_owned()),
+            bank_id: context.get_identifier(Self::BANK.to_owned()),
+            transmitter,
+            receiver,
+            alt_above_ground: Length::new::<foot>(0.),
+            pitch: Angle::new::<degree>(0.),
+            bank: Angle::new::<degree>(0.),
+        }
+    }
+
+    fn transmitter_height_over_ground(&self) -> Length {
+        let vertical_offset = self.pitch.sin() * self.transmitter.z;
+        self.alt_above_ground + vertical_offset - self.transmitter.y
+    }
+
+    fn receiver_height_over_ground(&self) -> Length {
+        let vertical_offset = self.pitch.sin() * self.receiver.z;
+        self.alt_above_ground + vertical_offset - self.receiver.y
+    }
+
+    fn distance_between_antennas(&self) -> Length {
+        (self.transmitter.z - self.receiver.z).abs()
+    }
+
+    fn antenna_along_ground_distance(&self) -> Length {
+        self.pitch.cos() * self.distance_between_antennas()
+    }
+}
+
+impl TransceiverPair for Ala52BTransceiverPair {
+    /// Returns the time the signal took to travel from one of the transceivers to the other,
+    /// to the receiver, or None if there is no path (or the signal is too weak).
+    /// While it is simplified compared to a proper frequency simulation, you can imagine that the
+    /// measured runtime is already based on the filtered difference between the sent and received
+    /// frequency in an FMCW-based radar.
+    fn response(&self) -> Option<TransceiverPairResponse> {
+        if self.pitch.abs() > Angle::new::<degree>(45.)
+            || self.bank.abs() > Angle::new::<degree>(44.)
```

Comment:
Is this a hard limit in the RA software, or approximated as the point where the return is too weak?

Reply:
No, these were just picked randomly with a good margin outside of the specified operating range. I presume it's mostly a factor of the ground material and incidence angle. In future we could model the terrain you're over to vary these, but the sim unfortunately only provides the terrain exactly below you, and I need to think about how we'll make it feel good rather than randomly disappearing when you fly over a tiny patch of trees. I didn't want to grow the scope too much though.

Reply:
Yep, fair enough. I was thinking maybe of multiplying the cosine of the two angles and a factor for the square of distance to give a rough approximation of return strength... but you'd end up tuning those to achieve something similar to this anyway. Good for initial impl. at the very least. 👍

Reply:
So I started doing that but I cut myself short and removed it again just because I didn't want to grow the scope.
Exactly, we can easily add this in future and this PR puts the architecture in place to update the system transparently once we want to spend more time on it.